File: app/src/main/java/com/konopko/stocktest/ApiService.java
package com.konopko.stocktest;

import com.konopko.stocktest.model.ModelTickerChart;
import com.konopko.stocktest.model.ModelTickerDetails;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {

    // Fetch ticker details
    @GET("stock/get-detail?region=US&lang=en")
    Single<ModelTickerDetails.Result> fetchTickerDetails(@Query("symbol") String tickerId);

    // Fetch ticker chart
    @GET("stock/v2/get-chart?interval=1d&region=US&lang=en&range=1mo")
    Single<ModelTickerChart.Result> fetchTickerChart(@Query("symbol") String tickerId);

}

File: app/src/main/java/com/konopko/stocktest/network/ApiService.kt
package com.konopko.stocktest.network

import com.konopko.stocktest.model.ModelTickerChart
import com.konopko.stocktest.model.ModelTickerDetails
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query


interface ApiService {

    // Fetch ticker details
    @GET("stock/get-detail?region=US&lang=en")
    fun fetchTickerDetails(@Query("symbol") tickerId: String): Single<ModelTickerDetails.Result>

    // Fetch ticker chart
    @GET("stock/v2/get-chart?interval=1d&region=US&lang=en&range=1mo")
    fun fetchTickerChart(@Query("symbol") tickerId: String): Single<ModelTickerChart.Result>
}